client/certificate_manager.py

```diff
@@ -0,0 +1,68 @@
+import os
+from datetime import date
+
+from PIL import Image, ImageDraw, ImageFont
+from bidi.algorithm import get_display
+from django.contrib.staticfiles import finders
+
+CERTIFICATE_TEXT_COLOR = (3, 8, 12)  # (R, G, B)
+CERTIFICATE_TEXT_SIZE = 40
+CERTIFICATE_TEXT_POSITION = (700, 200)
+
+
+def create_certificate_image(certificate):
+    """
+    Creates the volunteer certificate image with the current data and stores it under the correct path, overwriting
+    the possibly-existing file.
+    :param certificate: the VolunteerCertificate object to create the image for
+    """
+    volunteer = certificate.volunteer
+    tag_filename = finders.find('client/tag.jpeg')
+    font_filename = finders.find('client/fonts/BN Amnesia.ttf')
+    photo = None
+    try:
+        photo = Image.open(tag_filename)
+        drawing = ImageDraw.Draw(photo)
+        font = ImageFont.truetype(font_filename, size=CERTIFICATE_TEXT_SIZE)
+
+        lines_to_insert = [
+            f'שם מתנדב: {volunteer.first_name} {volunteer.last_name}',
+            f'תעודת זהות: {volunteer.tz_number}',
+            f'תוקף התעודה: {certificate.expiration_date}',
+            f'מספר תעודה: {certificate.id}',
+        ]
+
+        drawing.text(
+            CERTIFICATE_TEXT_POSITION,
+            get_display('\n'.join(lines_to_insert)),
+            fill=CERTIFICATE_TEXT_COLOR,
+            font=font,
+            align='right'
+        )
+        photo.save(certificate.image_path, format='png')
+    finally:
+        if photo is not None:
+            photo.close()
+
+
+g_last_certificate_purge = None
```

Comment:
there are some problems with globals in django. you are not guarnteed that they are kept between requests, especially in production.

Reply:
It probably should be managed by the certificate model ('cached_certificate_image' field or something like that)